sdk/openai/Azure.AI.OpenAI/api/Azure.AI.OpenAI.netstandard2.0.cs

```diff
@@ -1,3 +1,16 @@
+namespace Azure
+{
+    public partial class StreamingResponse<T> : System.Collections.Generic.IAsyncEnumerable<T>, System.IDisposable
+    {
+        internal StreamingResponse() { }
+        public static Azure.StreamingResponse<T> CreateFromResponse(Azure.Response response, System.Func<Azure.Response, System.Collections.Generic.IAsyncEnumerable<T>> asyncEnumerableProcessor) { throw null; }
```

Comment:
Review note: having a resolver delegate instead of just providing the `IAsyncEnumerable` directly is an extra level of perhaps unnecessary indirection, but it just felt very "off" to have no direct connection between the response and the enumerable; e.g. with direct use, you could do this:
```csharp
StreamingResponse<Foo> response = StreamingResponse<Foo>.CreateFromResponse(
    response,
    new Foo[] { new Foo("ha, what response? I enumerate whatever I want!") }.ToAsyncEnumerable());
```
Having a delegate that's invoked against the provided response at least makes it considerably harder to accidentally do things like the above. I'd prefer to have the resolver built into the type (`Foo`), but without something along the lines of C# 11 `static abstract` interface methods, that doesn't seem to have an elegant approach available.